cmd/skopeo/sync.go

```diff
@@ -723,6 +726,19 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 		logrus.Warn("Running in dry-run mode")
 	}
 
+	var digestFile *os.File
+	if opts.digestFile != "" && !opts.dryRun {
+		digestFile, err = os.OpenFile(opts.digestFile, os.O_CREATE|os.O_WRONLY, 0644)
```

Comment:
Without `O_APPEND`, this now needs `O_TRUNC`, otherwise the file will contain data from a previous run.